### PR TITLE
Task # AST-31546

### PR DESCRIPTION
**Pull request by margaritalm**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

> Describe the purpose of this PR along with any background information and the impacts of the proposed change.

### References

> Include supporting link to GitHub Issue/PR number

### Testing

> Describe how this change was tested. Be specific about anything not tested and reasons why. If this solution has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors.
>
> Please include any manual steps for testing end-to-end or functionality not covered by unit/integration tests.

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).
- [ ] I have updated the CLI help for new/changed functionality in this PR (if applicable).
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

File: internal/commands/result.go [hunk 1]
```diff
@@ -1081,6 +1089,53 @@ func exportGlSastResults(targetFile string, results *wrappers.ScanResultsCollect
...
+	createdAt, err := time.Parse(summaryCreatedAtLayout, summary.CreatedAt)
+	if err != nil {
+		return err
+	}
+
+	glDependencyResult.Schema = "https://gitlab.com/gitlab-org/gitlab/-/raw/master/lib/gitlab/ci/parsers/security/validators/schemas/15.0.0/sast-report-format.json"
+	glDependencyResult.Version = "15.0.0"
+	glDependencyResult.Scan.Analyzer.VendorGlSCA.VendorGlname = wrappers.AnalyzerScaID
+	glDependencyResult.Scan.Analyzer.Name = wrappers.AnalyzerScaID
+	glDependencyResult.Scan.Analyzer.Name = wrappers.AnalyzerScaID
```
Comment (tamarleviCm) [inline internal/commands/result.go:1123]: duplicate line: glDependencyResult.Scan.Analyzer.Name = wrappers.AnalyzerScaID

File: internal/commands/result.go [hunk 2]
```diff
@@ -1429,6 +1499,107 @@ func parseGlSastVulnerability(result *wrappers.ScanResult, glSast *wrappers.GlSa
...
+			Links:       collectDependencyPackageLinks(result),
+			TrackingDep: wrappers.TrackingDep{
+				Items: collectDependencyPackageItemsDep(result),
+			},
+			Flags: make([]string, 0),
+			LocationDep: wrappers.GlDepVulnerabilityLocation{
+				File: *(result.ScanResultData.ScaPackageCollection.Locations[0]),
+				Dependency: wrappers.DependencyLocation{
+					Package:                   wrappers.PackageName{Name: result.ScanResultData.PackageIdentifier},
+					DependencyLocationVersion: "",
```
Comment (tamarleviCm) [inline internal/commands/result.go:1522]: you can delete these empty properties, I think it is the same if you don't provide them

**Review by tamarleviCm — COMMENTED** [review]
you have to add an unit tests and integration tests

File: internal/commands/result.go [hunk 3]
```diff
@@ -1374,6 +1429,21 @@ func convertCxResultToGlVulnerability(results *wrappers.ScanResultsCollection, g
 	}
 }
 
+func convertCxResultToGlDependencyVulnerability(results *wrappers.ScanResultsCollection, glDependencyResult *wrappers.GlDependencyResultsCollection, summaryBaseURI string) {
```
Comment (OrShamirCM) [inline internal/commands/result.go:1432]: Please create a differnet go file that contain only GL logic and call if from the results

**Review by OrShamirCM — CHANGES_REQUESTED** [review]
Please add unit tests for the gl new file and logic + integration test from results for "sad" + "happy" flow

**Pull request closed without merging**